### ISSUE TITLE
After enabling ANR detection for the main process, all Electron processes are spawned twice

### ISSUE THREAD

**Issue by beyondkmp** [getsentry/sentry-electron#784]
### Is there an existing issue for this?

- [X] I have checked for existing issues https://github.com/getsentry/sentry-javascript/issues
- [X] I have reviewed the documentation https://docs.sentry.io/
- [X] I am using the latest SDK release https://github.com/getsentry/sentry-javascript/releases

### How do you use Sentry?

Sentry Saas (sentry.io)

### Electron SDK Version

4.14.0

### Electron Version

27.0.3

### What platform are you using?

Windows

### Steps to Reproduce

After enabling ANR detection for the main process, all Electron processes are being repeatedly created.

![image](https://github.com/getsentry/sentry-electron/assets/5462444/fd63bc3c-941b-4f7c-a499-8745685b425b)

This ANR detection relies on the sentry/node ANR module, which checks and spawns a child process during application startup. However, this approach is causing other child processes to be duplicated in Electron.

Below is my code. Could you please help me check whether the issue is with my code or if this feature itself has a problem?
https://github.com/beyondkmp/Sentry-electron-anr

### Expected Result

Only one additional ANR detection process is spawned

### Actual Result

All processes are spawned twice

**Comment by fredyangRC** [comment]
The ANR monitoring was a feature we eagerly want to have. 
Big thanks to the team effort that made that happen! However we also got a blocking issue.
I hope that you can help facilitate the priority for this fix.

Thanks,
Fred

**Comment by lforst** [comment]
Thanks for reporting this. We will take a look!

**Comment by timfish** [comment]
> this approach is causing other child processes to be duplicated in Electron

How are you starting other child processes? I looked at the demo repo and couldn't find anything obvious.

**Comment by beyondkmp** [comment]
Yes, there are two electron instance. Then there are duplicate network/gpu/crashpad process.

**Comment by beyondkmp** [comment]
@timfish 
```
export function enableMainProcessAnrDetection(options: Parameters<typeof enableNodeAnrDetection>[0]): Promise<void> {
  if (ELECTRON_MAJOR_VERSION < 4) {
    throw new Error('Main process ANR detection is only supported on Electron v4+');
  }

  const mainOptions = {
    entryScript: app.getAppPath(),
    ...options,
  };

  return enableNodeAnrDetection(mainOptions);
}
```
We can add `ELECTRON_RUN_AS_NODE =1`, then this anr process is just a node process instead of multiple electron process.

**Comment by timfish** [comment]
There are issues with using a node process for ANR, rather than another electron main process:
- We currently run the existing app main entry point again with a flag. This code (including our SDK) imports `electron` which will fail in node mode.
- Without access to Electron APIs we can't resolve much of the app context including the default `appname@version` release name.

I'll be back from vacation shortly and I'll be able to look into this further.

**Comment by timfish** [comment]
`@sentry/node` now uses a worker thread rather than a child process. Once we get the Electron SDK using the latest JavaScript SDKs this upgrade should be supported in Electron too.

**Comment by timfish** [comment]
This was closed by #810 and will be in the next release.
